src/map/attack.cpp

```diff
@@ -428,19 +461,19 @@ void CAttack::ProcessDamage()
         m_damage = (uint32)(((m_baseDamage + m_naturalH2hDamage + m_trickAttackDamage +
             battleutils::GetFSTR(m_attacker, m_victim, GetWeaponSlot())) * m_damageRatio));
     }
-    // Not H2H.
-    else
+    else if (GetWeaponSlot() == SLOT_MAIN)
```

Comment:
we can store `GetWeaponSlot()` return value in a variable instead of calling it each time and repeating the checks in the function